pkg/validators/am0005_test_harness.go

```diff
@@ -0,0 +1,81 @@
+// ValidateAddonLabel validates whether the 'label' field under an addon.yaml follows the format 'api.openshift.com/addon-<id>'
+// TODO - remove this validator once we do field level validation
+package validators
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+	imageparser "github.com/novln/docker-parser"
+)
+
+const quayRegistryApi = "https://quay.io/v2"
+
+var AM0005 = utils.Validator{
+	Code:        "AM0005",
+	Name:        "test_harness",
+	Description: "Ensure that an addon has a valid testharness image",
+	Runner:      ValidateTestHarness,
+}
+
+func init() {
+	Registry.Add(AM0005)
+}
+
+func ValidateTestHarness(metabundle utils.MetaBundle) (bool, string, error) {
+	res, err := imageparser.Parse(metabundle.AddonMeta.TestHarness)
+	if err != nil {
+		return false, "Failed to parse testharness url", err
+	}
+	if res.Registry() != "quay.io" {
+		return false, "Testharness image is not in the quay.io registry", nil
+	}
+	imagePresent, err := checkImageExists(res, 5)
+	if err != nil {
+		return false, "Encountered an error when trying to access the remote image", err
+	}
+	if !imagePresent {
+		return false, "Test harness image doesnt exist", nil
+	}
+	return true, "", nil
+}
+
+func checkImageExists(imageUri *imageparser.Reference, retryCount int) (bool, error) {
+	apiUrl := fmt.Sprintf("%s/%s/manifests/%s", quayRegistryApi, imageUri.ShortName(), imageUri.Tag())
+	resp, err := http.Get(apiUrl)
+	if err != nil {
```

Comment:
Suggest a refactor:
- add `time.Sleep()` -> could get fancy and use `retryCount`
- simplify logic in one line
- `retryableError(err)` has 1 arg

```go
if err != nil && retryCount > 0 && retryableRequestError(err) {
  time.Sleep(2 * time.Second)
  return checkImageExists(imageUri, retryCount-1)
}
```

Reply:
If we simplify the logic in one line (`err != nil && retryCount > 0 && retryableRequestError(err) `) We'll have to add check `resp ` is not nil for all the lines below it which expect response to be non nil. Also, we still need to add a
branch which returns the error. 
What about this instead?
```
if err != nil {
	if retryCount > 0 && retryableError(err) {
		time.Sleep(2 * time.Second)
		return checkImageExists(imageUri, retryCount-1)
	}
	return false, err
}
```